File: Lab1.1/Lab1/Row.cs
// Оголошення класу Row
class Row
{
    // Публічне поле, що містить список символів
    public List<char> Values;

    // Конструктор, що приймає список символів і зберігає його у полі Values
    public Row(List<char> values)
    {
        Values = values;
    }

    // Метод, що повертає масив символів, створений з поля Values
    public char[] GetValues()
    {
        return Values.ToArray();
    }

    // Метод, що повертає кількість символів у списку
    public int GetLength()
    {
        return Values.Count;
    }

    // Метод, що видаляє ПЕРШЕ входження символу '5' із списку
    public void RemoveFive()
    {
        Values.Remove('5');  // Remove видаляє тільки перший знайдений елемент
    }
}

File: Lab1.1/Lab1CPP/Row.cpp
#include "Row.h"

// Метод повертає константне посилання на вектор символів (_values)
const std::vector<char>& Row::getValues() const { return _values; };

// Метод повертає кількість символів у векторі (_values)
size_t Row::getLength() const { return _values.size(); };

// Метод видаляє ПЕРШЕ входження символу '5' із вектора
void Row::removeFive() {
    for (int i = 0; i < _values.size(); i++) {
        if (_values[i] == '5') {
            _values.erase(_values.begin() + i);  // Видаляємо елемент на позиції i
            break;  // Після видалення зупиняємо цикл
        }
    }
}

